File: Laboratoria/Lab 01/Zadanie 04.c
#include <stdio.h>

// Funkcja obliczająca silnię z podanej liczby
int factorial(int number) {
    // Jeśli liczba jest mniejsza lub równa 2
    if(number <= 2) {
        // Zwróć zmienną number
        return number;
        // W przeciwnym wypadku
    } else {
        // Oblicz silnię od (number - 1)
        return number * factorial(number - 1);
    }
}

// Main
int main()  {
    // Zainicjalizowanie zmiennej number
    int number = 5;

    // Obliczenie silni ze zmiennej number
    // Zapisanie wyniku w zmiennej factorial_result
    int factorial_number = factorial(number);

    // Wyświetlenie wyniku silni z podanej liczby
    printf("Silnia z liczby: %d, wynosi: %d!", number, factorial_number);

    return 0;
}

File: Laboratoria/Lab 11/Zadanie 04.hs
-- Main
main :: IO()
-- Wyświetlenie wartości silni od liczby 5
main = print(factorial(5))

-- Definicja funkcji factorial
-- Definicja typu zwracanego przez funkcję
factorial :: (Integer) -> (Integer)
-- Definicja obliczeń silni dla podanej liczby w sposób rekurencyjny
factorial(n) = 
    -- Jeśli n jest mniejsze lub równe 1
    if n <= 1 then
        -- Zwróć n
        n
    -- W przeciwnym przypadku
    else
        -- Oblicz n razy silnia od (n - 1)
        n * factorial(n - 1)
